mypy/checkexpr.py

```diff
@@ -179,7 +179,10 @@ def visit_call_expr(self, e: CallExpr, allow_none_return: bool = False) -> Type:
             return self.check_typeddict_call(e.callee.node.typeddict_type,
                                              e.arg_kinds, e.arg_names, e.args, e)
         self.try_infer_partial_type(e)
-        callee_type = self.accept(e.callee)
+        context_type = CallableType([self.accept(a) for a in e.args], e.arg_kinds, e.arg_names,
+                                    ret_type=self.type_context[-1] or AnyType(),
+                                    fallback=self.named_type('builtins.function'))
+        callee_type = self.accept(e.callee, context_type)
```

Comment:
Maybe it is too big a hammer to use on every call? should I add `if isinstance(e.callee, LambdaExpr):`?